superset/models/core.py

```diff
@@ -786,7 +786,12 @@ def grains(self):
         return self.db_engine_spec.time_grains
 
     def grains_dict(self):
-        return {grain.duration: grain for grain in self.grains()}
+        """Allowing to lookup grain by either label or duration
+
+        For backward compatibility"""
+        d = {grain.duration: grain for grain in self.grains()}
+        d.update({grain.label: grain for grain in self.grains()})
```

Comment:
How do the grains show up in the form? Do we have duplicates?

Reply:
Looks ok. The dropdown uses the `grains` property, not `grains_dict`.

Reply:
Awesome, thanks!